code/core/src/csf-tools/CsfFile.ts

```diff
@@ -544,14 +563,17 @@ export class CsfFile {
         if (play) {
           acc[key].tags = [...(acc[key].tags || []), 'play-fn'];
         }
-        ['play', 'render'].forEach((annotation) => {
-          acc[key].__stats[annotation as keyof IndexInputStats] =
+        const stats = acc[key].__stats;
+        ['play', 'render', 'beforeEach'].forEach((annotation) => {
```

Comment:
shouldn't we just get all used annotations? would make this kind of work unnecessary in the future as it would just include them, plus we get more info over usage of loaders, decorators, etc.